rogue/common/update_checker.py

```diff
@@ -0,0 +1,256 @@
+    }
+
+    try:
+        with open(cache_file, "w") as f:
+            json.dump(cache_data, f, indent=2)
+    except IOError:
+        pass  # Silently handle write errors
+
+
+def _is_newer_version(latest: str, current: str) -> bool:
+    """Compare version strings to determine if latest is newer than current."""
+    try:
+        return version.parse(latest) > version.parse(current)
+    except version.InvalidVersion:
+        # Handle non-standard version formats gracefully
+        return False
+
+
+def _show_update_prompt(latest_version: str, current_version: str) -> None:
+    """Display the update prompt with rich formatting and interactive option."""
+    console = Console()
+
+    # Create the update message
+    title = Text("🚀 Update Available!", style="bold yellow")
+
+    content = Text()
+    content.append("A new version of rogue-ai is available!\n\n", style="")
+    content.append("Current version: ", style="dim")
+    content.append(f"{current_version}\n", style="red")
+    content.append("Latest version:  ", style="dim")
+    content.append(f"{latest_version}\n\n", style="green bold")
+    content.append("To update manually, run: ", style="dim")
+    content.append("uv tool upgrade rogue-ai", style="cyan bold")
+    content.append(" or ", style="dim")
+    content.append("uvx --refresh rogue-ai", style="cyan bold")
+
+    # Create a panel with the update message
+    panel = Panel(
+        content,
+        title=title,
+        border_style="yellow",
+        padding=(1, 2),
+    )
+
+    # Print with some spacing
+    console.print()
+    console.print(panel)
+
+    # Ask user if they want to update now
+    try:
+        should_update = Confirm.ask(
+            "[bold cyan]Would you like to update now?[/bold cyan]",
+            default=True,
+        )
+
+        if should_update:
+            run_update_command()
+        else:
+            console.print(
+                "[dim]Update skipped. Run 'uv tool upgrade rogue-ai' or "
+                "'uvx --refresh rogue-ai' later to update.[/dim]",
+            )
+    except (KeyboardInterrupt, EOFError):
+        # Handle Ctrl+C or input interruption gracefully
+        console.print("\n[dim]Update skipped.[/dim]")
+
+    console.print()
+
+
+def run_update_command() -> None:
+    """Execute the appropriate update command based on installation method."""
+    console = Console()
+
+    try:
+        console.status(
+            "[yellow]Updating rogue-ai...[/yellow]",
+            spinner="dots",
+        )
+        console.print(
+            "[dim]This may take a few minutes to download and install "
+            "dependencies...[/dim]",
+        )
+
+        if not shutil.which("uv"):
+            console.print(
+                "[dim]uv not found. please update manually using[/dim]"
+                "[dim]- uv tool upgrade rogue-ai[/dim]"
+                "[dim]or[/dim]"
+                "[dim]- pip install -e . (if you cloned the repo)[/dim]",
```

Comment:
if the user cloned the repo and pulled some changes, then `pip -E` will automatically load the changes anywhays. I think we should just use this instead here:
```suggestion
                "[dim]- pip install rogue-ai -U[/dim]",
```